botocore/awsrequest.py

```diff
@@ -534,6 +557,84 @@ def reset_stream(self):
             raise UnseekableStreamError(stream_object=self.body)
 
 
+class AWSRequestCompressor:
```

Comment:
The spec describes this concept as "request compression" and the trait is known as `requestcompression` so it seemed logical to associate this with the entire object of `AWSRequest` rather than a specific component of it.